docker/substra-backend/Dockerfile

```
@@ -1,19 +1,23 @@
-FROM python:3.11-slim AS build
+FROM ubuntu:23.10 AS build
 
 ENV GRPC_POLL_STRATEGY=epoll1
 ENV PYTHONUNBUFFERED=1
 
-ARG DEBIAN_FRONTEND=noninteractive
-
-RUN apt-get update && apt-get upgrade -y && apt-get install -y --no-install-recommends \
-    build-essential \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN apt-get update -y \
+    && apt-get install --no-install-recommends -y --only-upgrade gnupg \
+    && apt-get install --no-install-recommends -y software-properties-common \
+    && apt-get --no-install-recommends -y install python3-pip gcc python3.11-dev python3-watchdog celery python-is-python3 python3.11-venv \
```

we could merge these two lines
```suggestion
    && apt-get install --no-install-recommends -y software-properties-common \
    python3-pip gcc python3.11-dev python3-watchdog celery python-is-python3 python3.11-venv \
```